blog/tests/test_views.py

```diff
@@ -0,0 +1,101 @@
+        self.client = Client()
+        self.user = User.objects.create_superuser(
+            username=self.USERNAME, password=self.PASSWORD,
+            email=self.EMAIL,
+        )
+
+    def test_index_view_rendering(self):  # this is a test case
+        """Testing main page for render needed posts."""
+        tz = timezone.get_current_timezone()
+        post = Post.objects.create(author=self.user, title='Test', text='superText',
+                                   created_date=datetime(day=1, month=3, year=2016, tzinfo=tz),
+                                   published_date=datetime(day=1, month=3, year=2016, tzinfo=tz))
+        past_post = Post.objects.create(author=self.user, title='past_test', text='superText',
+                                        created_date=datetime(day=1, month=4, year=2015, tzinfo=tz),
+                                        published_date=datetime(day=1, month=4, year=2015, tzinfo=tz))
+        future_post = Post.objects.create(author=self.user, title='future_test', text='superText',
+                                          created_date=datetime(day=1, month=4, year=2116, tzinfo=tz),
+                                          published_date=datetime(day=1, month=4, year=2116, tzinfo=tz))
+        with patch('django.utils.timezone.now', lambda: datetime(day=1, month=1, year=2016, tzinfo=tz)):
+            response = self.client.get(reverse('post_list'))
+            self.assertListEqual(list(response.context['posts']), [past_post])
+            self.assertNotContains(response, post)
+            self.assertNotContains(response, future_post)
+        with patch('django.utils.timezone.now', lambda: datetime(day=1, month=4, year=2016, tzinfo=tz)):
+            response = self.client.get(reverse('post_list'))
+            self.assertListEqual(list(response.context['posts']), [post, past_post])
+            self.assertNotContains(response, future_post)
+        with patch('django.utils.timezone.now', lambda: datetime(day=1, month=4, year=3016, tzinfo=tz)):
+            response = self.client.get(reverse('post_list'))
+            self.assertListEqual(list(response.context['posts']), [future_post, post, past_post])
+        response = self.client.get(reverse('post_list'))
+        self.assertEqual(200, response.status_code)
+        self.assertTemplateUsed(response, 'blog/base.html')
+        self.assertContains(response, post)
+        self.assertContains(response, past_post)
+        self.assertNotContains(response, future_post)
+
+    def test_detail_view(self):
+        """Testing detail page when post is not exist and when it exists."""
+        response = self.client.get(reverse('post_detail', kwargs={'pk': 1}))
+        self.assertEqual(404, response.status_code)
+        post = Post.objects.create(author=self.user, title='Test', text='superText')
+        response = self.client.get(reverse('post_detail', kwargs={'pk': post.pk}))
+        self.assertEqual(200, response.status_code)
```

Comment:
Did the post actually appear on the web page?

Reply:
So?

Reply:
Done